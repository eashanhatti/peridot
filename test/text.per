metadefine foo: Text = "abc"
metadefine _: Text = foo ++ "bar"