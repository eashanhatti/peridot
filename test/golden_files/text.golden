((fromList [], fromList [], fromList []),
 Declare
   Meta
   (Rigid
      (RNameIntro
         (UserName "<PLACEHOLDER>") Meta (Id {unId = 0})))
   (Rigid TextType)
   (Declare
      Meta
      (Rigid
         (RNameIntro
            (UserName "<PLACEHOLDER>") Meta (Id {unId = 1})))
      (Rigid TextType)
      (Define
         (Rigid
            (RNameIntro
               (UserName "<PLACEHOLDER>") Meta (Id {unId = 0})))
         (Rigid
            (TextIntroCons
               'a'
               (Rigid
                  (TextIntroCons
                     'b'
                     (Rigid (TextIntroCons 'c' (Rigid TextIntroNil)))))))
         (Define
            (Rigid
               (RNameIntro
                  (UserName "<PLACEHOLDER>") Meta (Id {unId = 1})))
            (TextElimCat
               (GlobalVar
                  (Rigid
                     (RNameIntro (UserName "foo") Meta (Id {unId = 0})))
                  False)
               (Rigid
                  (TextIntroCons
                     'b'
                     (Rigid
                        (TextIntroCons
                           'a'
                           (Rigid (TextIntroCons 'r' (Rigid TextIntroNil))))))))
            (Rigid (TypeType Obj))))))
